src/main/java/com/emc/ecs/servicebroker/service/EcsService.java

```diff
@@ -912,6 +914,23 @@ public Map<String, Object> changeNamespacePlan(String namespace, ServiceDefiniti
                 }
             }
         }
+
+        @SuppressWarnings("unchecked")
+        Map<String, Integer> quota = (Map<String, Integer>) parameters.getOrDefault(QUOTA, new HashMap<>());
+        int limit = quota.getOrDefault(QUOTA_LIMIT, -1);
+        int warn = quota.getOrDefault(QUOTA_WARN, -1);
+
+        if (limit == -1 && warn == -1) {
+            logger.info("Removing quota from namespace '{}'", prefix(namespace));
+            NamespaceQuotaAction.delete(connection, prefix(namespace));
```

Comment:
will `NamespaceQuotaAction.delete` fail if there was no quota to begin with?

Reply:
The api does not fail even when quota was set before previously.
We have tested below scenarios: 

| From | To |
|-------|----|
| Quota 1 | Quota 2 |
| Quota 2 | Unlimited |
| Unlimited  | Unlimited |
| Unlimited  | Quota 1 |


We have also verified this by calling the management api via Postman.